frontend/gamelib.js

```diff
@@ -267,9 +270,13 @@ class Region {
 	}
 
 	calcNextHarvest() {
-		let base = new Calc("*", [
-			{"v": this.population, "unit": " citizens", "why": "Regional Population"},
-			{"v": 13, "unit": " plants / citizen", "why": "Global Planting Rate"}]);
+		let mods = [];
+		if (this.calcNobleLevel() > 0) mods.push({"v": this.calcNobleLevel() * .05, "unit": "%", "why": "Noble"});
+		let base = Calc.moddedNum(
```

Comment:
This comment is not really specifically about this PR but now that I've found where those calculation tooltips are created.  A suggestion I have as a player and frequent consumer of these tooltips is to format that numbers in the tooltips to be a little shorter.  Seeing numbers like 13.37842748623746873 several times makes it difficult to read and understand.

Reply:
Another wider-scope suggestion, maybe consider using this same Calc tooltip builder to implement what I think of as the "function" rules of the game, for example how to compute unrest effects on recruitment.  You already have the object notation for input, could expand its usage to the backend.

Reply:
re: the first comment, do you have an example place where that happens in mind?

The rounding behavior is encapsulated in explainArg, which cuts the number at (up to) five decimal places. However, there still are long numbers, especially around population, size, food, etc, where the significant figures are left of the decimal point; it's less clear to me what to do with those (e.g. rendering 13000000 doesn't seem inherently better than 13378427, and I think you're in the minority of preferring 1.3E7; maybe what is needed is 13,378,427?)

---

re: the second comment, yes, that's kind of what I am getting at with the idea of a grammar that explains the game. In my mind, the main issue with making Calc language-independent is the matter of references (e.g. how to express the "region.population" part in "sqrt(region.population)"?) - to do that right, it basically suggests creating a system that describes the entities in the game and their properties and relationships, which starts to look like it approaches a full declarative programming language.